### PR TITLE
Add block properties - Closes #703

### LINKED ISSUE #703: Add block properties to postBlock
Module transport receives reduced blocks through `postBlock` internal API. Should call `addBlockProperties` before normalize block.

### PR DESCRIPTION
**Pull request by LucasIsasmendi**
Add default block properties when receives a reduced block

### CODE REVIEW

**Comment by Isabello** [comment]
This PR does not solve the issues with block propagation. I have relayed the information where to start fixing it.

**Comment by Isabello** [comment]
The new commits resolve the issues I was seeing. I think we should look to remove the `addBlockProperties` and `removeBlockProperties` functions. While reducing the size of the block is a positive thing, mutating the block before and after receipt is not in our best interest. We should be processing it as one package on all nodes.

**Comment by Isabello** [comment]
```
  postBlock
> Error / Response: "TypeError: Cannot read property 'version' of undefined" undefined
>> /home/lisk/workspace/lisk-core_PR-718-TECICBJVR7R6W66C3HXZMSSUKBRF5YRBWNQWIEG6UBVXT3I6VABA/node_modules/js-nacl/lib/nacl_factory.js:27
>> var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=false;var ENVIRONMENT_IS_WORKER=false;var ENVIRONMENT_IS_NODE=false;var ENVIRONMENT_IS_SHELL=false;if(Module["ENVIRONMENT"]){if(Module["ENVIRONMENT"]==="WEB"){ENVIRONMENT_IS_WEB=true}else if(Module["ENVIRONMENT"]==="WORKER"){ENVIRONMENT_IS_WORKER=true}else if(Module["ENVIRONMENT"]==="NODE"){ENVIRONMENT_IS_NODE=true}else if(Module["ENVIRONMENT"]==="SHELL"){ENVIRONMENT_IS_SHELL=true}else{throw new Error("The provided Module['ENVIRONMENT'] value is not valid. It must be one of: WEB|WORKER|NODE|SHELL.")}}else{ENVIRONMENT_IS_WEB=typeof window==="object";ENVIRONMENT_IS_WORKER=typeof importScripts==="function";ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="fu
>> AssertionError: expected 'TypeError: Cannot read property \'version\' of undefined' to include 'Failed to validate block schema'
>>   at Proxy.include (/home/lisk/workspace/lisk-core_PR-718-TECICBJVR7R6W66C3HXZMSSUKBRF5YRBWNQWIEG6UBVXT3I6VABA/node_modules/chai/lib/chai/core/assertions.js:508:10)
>>   at Proxy.chainableMethodWrapper (/home/lisk/workspace/lisk-core_PR-718-TECICBJVR7R6W66C3HXZMSSUKBRF5YRBWNQWIEG6UBVXT3I6VABA/node_modules/chai/lib/chai/utils/addChainableMethod.js:113:49)
>>   at Immediate.<anonymous> (/home/lisk/workspace/lisk-core_PR-718-TECICBJVR7R6W66C3HXZMSSUKBRF5YRBWNQWIEG6UBVXT3I6VABA/test/api/transport/transport.blocks.js:153:24)
>>   at runCallback (timers.js:674:20)
>>   at tryOnImmediate (timers.js:645:5)
>>   at processImmediate [as _immediateCallback] (timers.js:617:5)
>> Exited with code: 7.
>> Error executing child process: Error: Process exited with code 7.
Warning: Task "exec:coverageSingle" failed. Use --force to continue.
```

**Comment by Isabello** [comment]
Tests are failing when merged against 1.0.0. This is most likely because version is deleted before this point. Failure is here in test file: https://github.com/LiskHQ/lisk/blob/7eef2b3c9d8a2100db1320623599e8eab3c19a4b/test/api/transport/transport.blocks.js#L150

File: logic/broadcaster.js [hunk 1]
```diff
@@ -155,10 +153,12 @@ Broadcaster.prototype.broadcast = function (params, options, cb) {
 			}
 		},
 		function sendToPeer (peers, waterCb) {
-			if (options.data.block) {
-				options.data.block = bson.deserialize(options.data.block);
+			var optionDeserialized = JSON.parse(JSON.stringify(options));
```
Comment (4miners) [inline logic/broadcaster.js]: should be in try... catch block

**Pull request merged**